### ISSUE TITLE
High memory consumption when converting Flux<ByteBuf> to Mono<String>

### ISSUE THREAD

**Issue by kushagraThapar** [Azure/azure-sdk-for-java#5053]
**Describe the bug**
High memory consumption when converting Flux<ByteBuf> to Mono<String>

***Exception or Stack Trace***
No stack trace. 

**To Reproduce**
N/A

***Code Snippet***
```
public static Mono<String> toString(Flux<ByteBuf> contentObservable) {
        return contentObservable
                .reduce(
                        new ByteArrayOutputStream(INITIAL_RESPONSE_BUFFER_SIZE),
                        (out, bb) -> {
                            try {
                                bb.readBytes(out, bb.readableBytes());
                                return out;
                            } catch (IOException e) {
                                throw new RuntimeException(e);
                            }
                        })
                .map(out -> new String(out.toByteArray(), StandardCharsets.UTF_8));
    }
```

**Expected behavior**
Use Reactor Netty inbound receive with aggregation to get body as Mono<String> from channel

**Comment by JonathanGiles** [comment]
cc @jianghaolu

**Comment by kushagraThapar** [comment]
Released in V3.3.0
